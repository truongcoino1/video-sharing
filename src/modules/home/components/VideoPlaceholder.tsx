import React, { memo } from "react";

type VideoPlaceholderProps = {
  thumbnail: string;
  handleClick: () => void;
};
const VideoPlaceholder = ({
  thumbnail,
  handleClick,
}: VideoPlaceholderProps) => {
  return (
    <div className={`overflow-hidden relative`} onClick={handleClick}>
      <img
        className="w-full h-auto object-cover"
        src={thumbnail}
        alt="video placeholder"
        width="640"
        height="480"
      />
    </div>
  );
};

export default memo(VideoPlaceholder);
