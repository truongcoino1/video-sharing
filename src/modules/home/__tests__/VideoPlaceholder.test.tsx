import { render, cleanup } from "@testing-library/react";
import VideoPlaceholder from "../components/VideoPlaceholder";

describe("VideoPlaceholder", () => {
  afterEach(cleanup);

  it("VideoPlaceholder should render without crash", () => {
    const result = render(
      <VideoPlaceholder
        thumbnail={"https://i.ytimg.com/vi/r3ARBhMk_g4/sddefault.jpg"}
        handleClick={() => {}}
      />
    );
    const elm = result.container.querySelector(".video-placeholder");
    expect(elm).toBeInTheDocument();
  });

  it("VideoPlaceholder should render with correct thumbnail", () => {
    const result = render(
      <VideoPlaceholder
        thumbnail={"https://i.ytimg.com/vi/r3ARBhMk_g4/sddefault.jpg"}
        handleClick={() => {}}
      />
    );
    const elm = result.container.querySelector("img");
    expect(elm).toHaveAttribute(
      "src",
      "https://i.ytimg.com/vi/r3ARBhMk_g4/sddefault.jpg"
    );
  });
});
