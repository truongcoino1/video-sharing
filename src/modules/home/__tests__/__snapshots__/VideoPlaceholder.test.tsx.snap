// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlaceholder should render correct VideoPlaceholder 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="video-placeholder cursor-pointer rounded-large flex items-center justify-center overflow-hidden relative w-full lg:w-[350px] min-w-[350px] lg:h-[196px]"
      >
        <img
          alt="video placeholder"
          class="w-full lg:w-[350px] rounded-large lg:h-[196px] object-cover"
          height="480"
          src="https://i.ytimg.com/vi/r3ARBhMk_g4/sddefault.jpg"
          width="640"
        />
        <img
          alt="button play"
          class="absolute opacity-50"
          height="72"
          src="/youtube-brands.svg"
          width="72"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="video-placeholder cursor-pointer rounded-large flex items-center justify-center overflow-hidden relative w-full lg:w-[350px] min-w-[350px] lg:h-[196px]"
    >
      <img
        alt="video placeholder"
        class="w-full lg:w-[350px] rounded-large lg:h-[196px] object-cover"
        height="480"
        src="https://i.ytimg.com/vi/r3ARBhMk_g4/sddefault.jpg"
        width="640"
      />
      <img
        alt="button play"
        class="absolute opacity-50"
        height="72"
        src="/youtube-brands.svg"
        width="72"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
